packages/theming/demo/~patterns/stories/GetColorStory.tsx

```diff
@@ -0,0 +1,164 @@
+  const [perf, setPerf] = useState({ milliseconds: 0, calls: 0 });
+  const BASELINE_NOCACHE = 2780;
+  const BASELINE_CACHE = 2000;
+  let CALLS = 0;
+
+  const updateColor = (color: string) => {
+    setTimeout(() => {
+      setBackgroundColor(color);
+    }, 1);
+
+    CALLS++;
+  };
+
+  const variablesObject = theme.colors.variables[theme.colors.base];
+  const variables = Object.keys(variablesObject) as (keyof typeof variablesObject)[];
+  const hues = Object.keys(theme.palette).filter(hue => typeof theme.palette[hue] === 'object');
+  const offsets = [-300, -200, -100, 100, 200, 300];
+  const transparencies = Object.keys(theme.opacity);
+
+  const testGetColor = () => {
+    CALLS = 0;
+
+    variables.forEach(variable => {
+      const variableKeys = Object.keys(variablesObject[variable]);
+
+      variableKeys.forEach(variableKey => {
+        const parameters = { theme, variable: `${variable}.${variableKey}` };
+
+        updateColor(getColor(parameters));
+
+        transparencies.forEach(transparency => {
+          updateColor(getColor({ ...parameters, transparency: parseInt(transparency, 10) }));
+        });
+      });
+    });
+
+    hues.forEach(hue => {
+      updateColor(getColor({ theme, hue }));
+      updateColor(getColor({ theme, hue, dark: { hue } }));
+      updateColor(getColor({ theme, hue, light: { hue } }));
+
+      const shades = Object.keys(theme.palette[hue]);
+
+      shades.forEach(shade => {
+        const parameters = { hue, shade: parseInt(shade, 10) };
+
+        updateColor(getColor({ theme, ...parameters }));
+        updateColor(getColor({ theme, hue, dark: parameters }));
+        updateColor(getColor({ theme, hue, light: parameters }));
+
+        offsets.forEach(offset => {
+          updateColor(getColor({ theme, ...parameters, offset }));
+          updateColor(getColor({ theme, hue, dark: { ...parameters, offset } }));
+          updateColor(getColor({ theme, hue, light: { ...parameters, offset } }));
+        });
+
+        transparencies.forEach(_transparency => {
+          const transparency = parseInt(_transparency, 10);
+
+          updateColor(getColor({ theme, ...parameters, transparency }));
+          updateColor(getColor({ theme, hue, dark: { ...parameters, transparency } }));
+          updateColor(getColor({ theme, hue, light: { ...parameters, transparency } }));
+
+          offsets.forEach(offset => {
+            updateColor(getColor({ theme, ...parameters, transparency, offset }));
+            updateColor(getColor({ theme, hue, dark: { ...parameters, transparency, offset } }));
+            updateColor(getColor({ theme, hue, light: { ...parameters, transparency, offset } }));
+          });
+        });
+      });
+    });
+  };
+
+  const handleClick = () => {
+    setPerf({ milliseconds: 0, calls: 0 });
+
+    const startTime = performance.now();
+
+    testGetColor();
+    updateColor(initialColor);
+
+    const endTime = performance.now();
```

Comment:
This is captured after React has completed all state updates. `getColor`, on its own, is probably much faster than that! 🚀